packages/core/src/utils/stringUtils.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * Converts any value to a string. Stringifies functions and objects, making them ripe for hashing.
+ *
+ * Credit to Michael Wallace https://stackoverflow.com/a/59422590/12368615
+ *
+ * @param valueToConvert The value to convert
+ */
+export function anyToString(valueToConvert: unknown): string {
+  if (valueToConvert === undefined || valueToConvert === null) {
+    return valueToConvert === undefined ? 'undefined' : 'null';
+  }
+  if (typeof valueToConvert === 'string') {
+    return `'${valueToConvert}'`;
+  }
+  if (
+    typeof valueToConvert === 'number' ||
+    typeof valueToConvert === 'boolean' ||
+    typeof valueToConvert === 'function'
+  ) {
+    return valueToConvert.toString();
+  }
+  if (valueToConvert instanceof Array) {
+    const stringfiedArray = valueToConvert
+      .map((property) => anyToString(property))
+      .join(',');
+    return `[${stringfiedArray}]`;
+  }
+  if (typeof valueToConvert === 'object') {
+    const stringfiedObject = Object.entries(valueToConvert)
+      .map((entry: [string, unknown]) => {
+        return `${entry[0]}: ${anyToString(entry[1])}`;
+      })
+      .join(',');
+    return `{${stringfiedObject}}`;
+  }
+  return JSON.stringify(valueToConvert);
+}
```

Comment:
[nit] I see that you did it this way to deal with functions too, and that doesn't work with JSON.stringify, but I feel like there should be an easier way of doing this, by leaveraging lodash or something, maybe we can use JSON.stringify and provide a custom resolver function to deal with functions?

Reply:
I found this on stackoverflow
```js
JSON.stringify(obj,function(key, value){
    return (typeof value === 'function' ) ? value.toString() : value;
});
```

Reply:
ah nice! that's way more concise and seems to get the job done 👌